### ISSUE TITLE
Documentation Improvement

### ISSUE THREAD

**Issue by gabriel-simonetti** [gstt/laravel-achievements#29]
Hello to all users of the laravel-achievements package!

I'm currently writing a new version of the documentation and would like to hear feedback from everyone.

Please consider leaving a comment letting me know about:

- How did you like the current README.md file? Is it clear? Is it beginner-friendly?
- Did you experience any problems while using this package that the documentation wasn't clear enough?
- How did you like the examples provided in the current documentation?
- Do you have anything else to add?

Thank you very much!

**Comment by kgenly** [comment]
I just got started with this package today and had a fairly robust achievement system (about 40 achievements, although some are "chained") going in about an hour. The documentation was quite adequate and clear, and the examples were excellent and got me going with all the basics. I'm still expanding my system, so I'll keep commenting if I run into issues. :)

**Comment by kgenly** [comment]
Okay, here's a minor typo:

```
use App\Achievements\Have1000GoldOnTheBag;

$user->setProgress(new Have100GoldOnTheBag(), $user->amountOfGoldOnTheBag);
```

I'm almost certain you mean IN the bag rather than on?

On to more important things -- one thing I recall that DID trip me up about the docs was calling $user->unlockedAchievements() got me an instance of AchievementProgress. It tripped me up pretty badly when I could not do things like $achievement->name or even $achievement->achievement_detail->name in my view. Eventually through reading the code I was able to figure out that you've called it detail(), and that $achievement->detail->name did just fine. I think it would be worth showing that as an example.

**Comment by gabriel-simonetti** [comment]
That's interesting. Do you thing unlockedAchievements() should return instances of AchievementDetail instead?

I believe that what the documentation is lacking most is a way to display unlocked achievements. I have an example in a parallel repo but I have not linked it here.

**Comment by gabriel-simonetti** [comment]
And thanks for your feedback! Please let me know if you run into more issues while using this package.

**Comment by kgenly** [comment]
No, I think what it returns now is fine, I just think there should be an explicit example included showing how to access the details. Even just demoing {{ $achievement->detail->name }} once would be a helpful hint for newbies.

**Comment by gabriel-simonetti** [comment]
Hi @kgenly !

I'm currently preparing base templates to be used for teaching purposes. Are you able and would you be willing to share the templates you have created on your system?

**Comment by kgenly** [comment]
Hi again @gabriel-simonetti ! I'd be happy to share some -- did you have any in particular in mind?

Here's something for my user profile view:

```
@if ($achievements->count() > 0)
      <div class="achievements" id="achievements">
        <h3>{{$user->name}}'s achievements</h3>
        @foreach($achievements as $achievement)
        <div class="achievement">
          <img
          src="{{ asset('img/badges/'.strtolower(str_replace(' ', '_', $achievement->details->name)). '.png') }}"
          title="{{$achievement->details->name}}" alt="{{$achievement->details->name}} achievement badge"
          >
        </div>
        @endforeach
      </div>
      @endif
```

**Comment by HDVinnie** [comment]
I use this in my profile view

```
            <div class="well">
                <div class="row">
                    <div class="col-md-12 profile-footer">
                        {{ $user->username }} - {{ trans('user.recent-achievements') }}:
                        @foreach($user->unlockedAchievements() as $a)
                            <img src="/img/badges/{{ $a->details->name }}.png" data-toggle="tooltip" title=""
                                 height="50px" data-original-title="{{ $a->details->name }}">
                        @endforeach
                        <div class="col-xs-1 achievements"><i
                                    class="{{ config('other.font-awesome') }} fa-trophy text-success"></i><span>{{ $user->unlockedAchievements()->count() }}</span>
                        </div>
                    </div>
                </div>
            </div>
```

**Comment by gabriel-simonetti** [comment]
Thank you @HDVinnie and @kgenly for sharing your templates!

Do you have an example on how you're alerting your users for new Unlocks and Progresses?

**Comment by kgenly** [comment]
I have this same code in all my achievements:

```
public function whenUnlocked($progress)
    {
      $user = $progress->achiever;
      ActivityAlert::create([
        'user_id' => $user->id,
        'alertable_type' => get_class($progress),
        'alertable_id' => $progress->id,
        'icon' => 'trophy',
        'url' => route('user.profile', ['id'=> $user->username]).'#achievements',
        'activity' => 'Achievement unlocked! '. $progress->details->name
      ]);
    }
```

Of course activity alerts is a model I've set up for various kinds of notifications.

**Comment by HDVinnie** [comment]
Im using the SWAL alert example you provided.
